packages/engine/src/experiment/package/simple.rs

```diff
@@ -13,18 +13,39 @@ pub struct SimpleExperiment {
     config: SimpleExperimentConfig,
 }
 
-// Returns whether to stop experiment.
-fn finish_run(n_remaining: &mut isize) -> bool {
-    assert!(*n_remaining > 0, "n_remaining {} <= 0", *n_remaining);
-    *n_remaining -= 1;
-    *n_remaining == 0
-}
-
-struct StepProgress {
+struct SimProgress {
     n_steps: usize,
     stopped: bool,
 }
 
+struct SimQueue<'a> {
+    max_num_steps: usize,
+    pkg_to_exp: &'a mut ExpPkgCtlSend,
+    pending_iter: &'a mut (dyn Iterator<Item = (usize, &'a serde_json::Value)> + Send),
+    active: HashMap<SimulationShortId, SimProgress>,
+    finished: HashMap<SimulationShortId, SimProgress>,
+}
+
+impl<'a> SimQueue<'a> {
+    async fn start_sim_if_available(&mut self) -> Result<()> {
+        if let Some((sim_idx, changed_props)) = self.pending_iter.next() {
+            self.active
+                .insert(sim_idx as SimulationShortId, SimProgress {
```

Comment:
I would move this cast into the iterator construction, which would also make it more obvious that a `usize` is being cast to a smaller type.